### DISCUSSION TITLE
LAMBDA function

### DISCUSSION THREAD

**Discussion by BrianHung** [handsontable/hyperformula#1268] [Feature requests]
Wondering if there is any plan in supporting the LAMBDA function from Excel and Sheets
https://support.microsoft.com/en-us/office/lambda-function-bd212d27-1cd1-4321-a34a-ccbf254b8b67

Think it would be a good supplement to needing to write custom function plugins, and just allow the end user to write formulas they want.

**Comment by adrianszymanski89** [comment]
Hi @BrianHung 

Thank you for this suggestion. I agree that supporting the Lambda function might be a great improvement for Hyperformula. 

@sequba let us know what you think about the possibility of implementing it.

**Comment by sequba** [comment]
Yes, it is a very good idea, but we still have some bugs that we need to focus on before we could put our efforts into adding new functionalities.

**Reply by BrianHung** [reply]
Any clue on hard it would be to implement as a custom plugin? Difficult part would be taking the arguments, and then parsing and evaluating it. Wonder if it requires some hyperformula internals not currently exposed for this (?).

**Reply by sequba** [reply]
We didn't consider `LAMBDA` when we thought about custom functions. Neither formula parser nor evaluator is exposed.

**Comment by sequba** [comment]
More info: https://github.com/handsontable/hyperformula/discussions/1478#discussioncomment-11852868

**Comment by adjenks** [comment]
This is my number one request. Having LAMBDA would really simplify my sheets.

**Reply by AMBudnik** [reply]
Thank you for feedback @adjenks We'll update you once we add the function.
